kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/DeploymentUtils.java

```diff
@@ -177,6 +178,28 @@ public static void waitForDeploymentRunning(String namespaceName, String podName
                         && kubeClient().isDeploymentRunning(namespaceName, podName));
     }
 
+    /**
+     * Wait for expected replicas of a deployment to be running.
+     *
+     * @param namespaceName the namespace name
+     * @param deploymentName the deployment name
+     * @param expectedReplicas the expected replicas
+     * @param timeout the timeout
+     * @return the boolean
+     */
+    public static boolean waitForDeploymentRunning(String namespaceName, String deploymentName, int expectedReplicas, Duration timeout) {
+        LOGGER.info("Waiting for deployment: {} replicas of {}/{} to be running", expectedReplicas, namespaceName, deploymentName);
+        List<Pod> pods = kubeClient().listPods(namespaceName, kubeClient().getPodSelectorFromDeployment(namespaceName, deploymentName));
```

Comment:
could we fail earlier if `pods.size() != expectedReplicas`. It looks like logically this has to be the case. Then we could remove the `AtomicInteger`